### ISSUE TITLE
Python Driver throws error when slicing deque

### ISSUE THREAD

**Issue by binarycrayon** [rethinkdb/rethinkdb#5608]
collections.deque is the default data structure used by 'Cursor', but not correctly sliced when returning result. I'm submitting a pull request to slice with `itertools.islice`

```
Traceback (most recent call last):
  File "debug_query.py", line 4, in <module>
    print r.table('filer_reads_weekday_0').order_by(index=r.desc('size')).limit(5).run(conn)
  File "/net/homedirs/yuxue/Development/gitlab/dcue/rethinkdb/net.py", line 196, in __str__
    val_str = ', '.join(map(repr, self.items[:10]))
TypeError: sequence index must be integer, not 'slice'

```

**Comment by larkost** [comment]
I believe that this is already fixed in `next`, specifically with 4c84e336. That change is slated go out soon with the 2.3 release. I will be closing this as duplicate, but if I have missed something please comment and I will re-open this.

**Comment by binarycrayon** [comment]
I see, I downloaded driver post6 and had that issue. Should I just use the driver provided in next instead? 

I added pull request but perhaps I'll just close it too.
https://github.com/rethinkdb/rethinkdb/pull/5609

**Comment by danielmewes** [comment]
Thanks for the pull request @binarycrayon . You can close it, since a similar fix already exists.

The error happens when trying to turn a cursor into a string (for example for printing). As a work-around, you can convert it into a `list` first if it's small enough, and then print the list instead. Sorry for the inconvenience.

The issue that fixed this for the upcoming 2.3 driver was https://github.com/rethinkdb/rethinkdb/issues/5331

**Comment by binarycrayon** [comment]
Closed, thanks for the details @danielmewes !
